File: 10845.py
# 10845번: 큐
import sys


def push(que, num):
    que.append(num)


def pop(que):
    if len(que) == 0:
        print(-1)
    else:
        print(que[0])


def size(que):
    print(len(que))


def empty(que):
    if len(que) == 0:
        print(1)
    else:
        print(0)


def front(que):
    if len(que) == 0:
        print(-1)
    else:
        print(que[0])


def back(que):
    if len(que) == 0:
        print(-1)
    else:
        print(que[-1])


if __name__ == "__main__":
    ique = []  # 큐
    count = int(sys.stdin.readline())
    while count > 0:
        order = sys.stdin.readline().rstrip()
        if (order.split())[0] == "push":
            push(ique, order.split()[1])
        elif order == "pop":
            pop(ique)
            ique = ique[1:]
        elif order == "size":
            size(ique)
        elif order == "empty":
            empty(ique)
        elif order == "front":
            front(ique)
        elif order == "back":
            back(ique)
        count -= 1

File: 10845.cpp
// 11845번: 큐
#include <iostream>
#include <string>
#include <queue>
using namespace std;

queue<int> q;
int N; // 명령의 수
string order; // 명령을 저장하는 변수
int num; // push할때의 숫자를 저장하는 변수

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	cin >> N;
	while (N--) {
		cin >> order;
		if (order == "push") {
			cin >> num;
			q.push(num);
		}
		if (order == "pop") {
			if (q.empty()) {
				cout << -1 << "\n";
				continue;
			}
			cout << q.front() << "\n";
			q.pop();
		}
		if (order == "size")
			cout << q.size() << "\n";
		if (order == "empty") {
			cout << q.empty() << "\n";
		}
		if (order == "front") {
			if (q.empty()) {
				cout << -1 << "\n";
				continue;
			}
			cout << q.front() << "\n";
		}
		if (order == "back") {
			if (q.empty()) {
				cout << -1 << "\n";
				continue;
			}
			cout << q.back() << "\n";
		}

	}
	return 0;

}